### DISCUSSION TITLE
PWM with pcf8574

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#18079] [Q&A]
Can I use the outputs of a PCF8574 as PWM outputs in Tasmota?

**Comment by sfromis** [comment]
No

**Reply by barbudor** [reply]
But Tasmota supports PCA9685 which is a 16 channels PWM driver and the PCA9632 which is a 4 channels
None are build-in by default in Tasmota and would requires you compile your own binary

**Reply by ChrisVonFrosta** [reply]
Thank you! maybe i said the ic from the wrong (relais) board. i will have a look.
But very good to know!
thx

**Reply by ChrisVonFrosta** [reply]
it would be connectet

> No
Is there a hardware limitation preventing it, or is there simply no requirement or capability for it?

**Reply by barbudor** [reply]
PCF8574 is only capable of Hi and Low and doesn't have an internal PWM circuitry
In theory you could be able to send it commands to switch between Hi and Low bit that would use much of the CPU capability and have a very uncertain result due to the duration of the I2C commands.

If you really want a lot of PWM, go for the PCA9685

**Reply by ChrisVonFrosta** [reply]
"If you really want a lot of PWM, go for the PCA9685"
this looks pretty well. i should think about this option weeks before.
Thank you very muuch for your input!

**Reply by sfromis** [reply]
If you really want to get your fingers dirty, in principle it would not be impossible to grab the software PWM supported with ESP8266, and adapt it for sending bits to PCF8574, maybe not the best PWM rate. Not a recommended strategy, but if you want a challenge.....
